### PR TITLE
OCPBUGS-43316: Enforce privileged PSA by default

### PR DESCRIPTION
**Pull request by Evan-Reilly**
**What this PR does / why we need it**:
OpenShiftPodSecurityAdmission feature gate is not enabled by default in 4.17 any longer. Therefore, to ensure namespace security is not enforcing restricted by default any longer, this PR is required to properly enforce privileged PSA by default.

- https://github.com/openshift/api/blob/release-4.17/features/features.go
- https://github.com/openshift/api/pull/2018

**Which issue(s) this PR fixes** *(optional, use `fixes #<issue_number>(, fixes #<issue_number>, ...)` format, where issue_number might be a GitHub issue, or a Jira story*:
Fixes https://issues.redhat.com/browse/OCPBUGS-43316

**Checklist**
- [x] Subject and description added to both, commit and PR.
- [x] Relevant issues have been referenced.
- [ ] This change includes docs. 
- [ ] This change includes unit tests.

### CODE REVIEW

File: control-plane-operator/controllers/hostedcontrolplane/clusterpolicy/config.go [hunk 1]
```diff
@@ -67,5 +67,24 @@ func reconcileConfig(cfg *openshiftcpv1.OpenShiftControllerManagerConfig, minTLS
 			CipherSuites:  cipherSuites,
 		},
 	}
+
+	// disables automatically setting the `pod-security.kubernetes.io/enforce` label on namespaces by the pod-security-admission-label-synchronization-controller
+	// see https://github.com/openshift/cluster-policy-controller/blob/50c2a8337f08856bbae4cd419bb8ffcbdf92567c/pkg/cmd/controller/psalabelsyncer.go#L19
+	index := -1
+	for i := range cfg.FeatureGates {
+		fg := cfg.FeatureGates[i]
+		if strings.HasPrefix(fg, "OpenShiftPodSecurityAdmission") {
```
Comment (rtheis) [inline control-plane-operator/controllers/hostedcontrolplane/clusterpolicy/config.go:76]: @Evan-Reilly ptal at this failure report.

**Review by rtheis — COMMENTED** [review]
@Evan-Reilly please take a look at the failures

**Review by rtheis — APPROVED** [review]
/lgtm

**Review by rtheis — CHANGES_REQUESTED** [review]
/hold for changes based on https://ibm-argonauts.slack.com/archives/C01C8502FMM/p1727808157696799.

**Comment by rtheis** [comment]
@Evan-Reilly please fix the following test:

```
{Failed  === RUN   TestCreateClusterRequestServingIsolation/Main/EnsurePSANotPrivileged
    util.go:667: pod admitted when rejection was expected
    util.go:670: forbidden error expected, got %!s(<nil>)
        --- FAIL: TestCreateClusterRequestServingIsolation/Main/EnsurePSANotPrivileged (0.16s)
}
```

File: test/e2e/create_cluster_test.go [hunk 2]
```diff
@@ -129,7 +129,6 @@ func TestCreateClusterRequestServingIsolation(t *testing.T) {
 
 	e2eutil.NewHypershiftTest(t, ctx, func(t *testing.T, g Gomega, mgtClient crclient.Client, hostedCluster *hyperv1.HostedCluster) {
 		guestClient := e2eutil.WaitForGuestClient(t, testContext, mgtClient, hostedCluster)
-		e2eutil.EnsurePSANotPrivileged(t, ctx, guestClient)
```
Comment (rtheis) [inline test/e2e/create_cluster_test.go:132]: Per our slack discussion, I think you need to first remove the test in a separate PR in order for PR testing to pass on this PR.

**Review by rtheis — CHANGES_REQUESTED** [review]

File: test/e2e/create_cluster_test.go (hunk 2 above)
Comment (rtheis) [reply test/e2e/create_cluster_test.go:132]: https://github.com/openshift/hypershift/pull/4855 will hopefully resolve this.

**Comment by rtheis** [comment]
Retesting now that https://github.com/openshift/hypershift/pull/4875 merged.

File: control-plane-operator/controllers/hostedcontrolplane/clusterpolicy/config.go [hunk 3]
```diff
@@ -67,5 +67,14 @@ func reconcileConfig(cfg *openshiftcpv1.OpenShiftControllerManagerConfig, minTLS
 			CipherSuites:  cipherSuites,
 		},
 	}
+
+	if featureGates != nil {
```
Comment (rtheis) [inline control-plane-operator/controllers/hostedcontrolplane/clusterpolicy/config.go:71]: I don't think this is needed.  Similar to `cipherSuites` input, can't we simply set the value to what was passed into this function?

**Review by rtheis — CHANGES_REQUESTED** [review]
Evan please open a Jira for this PR too.

**Comment by rtheis** [comment]
/retitle OCPBUGS-43316: Enforce privileged PSA by default

**Review by rtheis — APPROVED** [review]
/lgtm

**Comment by rtheis** [comment]
@Evan-Reilly I think we may need some Jira magic done to make the tooling happy with this 4.17 only change.  We'll probably need Red Hat help to do this properly.

**Comment by rtheis** [comment]
@Evan-Reilly if my latest test retry fails, please contact the HyperShift folks for assistance.  There may be a known issue with this test.

**Comment by sjenning** [comment]
/hold

on review from hyperhift team.  didn't really no one from our side had looked yet.

File: control-plane-operator/controllers/hostedcontrolplane/clusterpolicy/params.go [hunk 4]
```diff
@@ -61,6 +63,16 @@ func (p *ClusterPolicyControllerParams) MinTLSVersion() string {
 	}
 }
 
+func (p *ClusterPolicyControllerParams) FeatureGates() []string {
```
[the remark this answers is not in the document]
Comment (Evan-Reilly) [reply control-plane-operator/controllers/hostedcontrolplane/clusterpolicy/params.go:66]: Yes, just pushed separate PR for that here: https://github.com/openshift/hypershift/pull/4925

**Pull request merged**